src/ParallelAlgorithms/Interpolation/Actions/InterpolatorReceiveVolumeData.hpp

```diff
@@ -65,7 +88,14 @@ struct InterpolatorReceiveVolumeData {
     tmpl::for_each<typename Metavariables::interpolation_target_tags>(
         [&holders, &this_temporal_id_is_done, &temporal_id](auto tag_v) {
           using tag = typename decltype(tag_v)::type;
-          if constexpr (std::is_same_v<TemporalId, typename tag::temporal_id>) {
+          // First off, this interpolation target must actually be using the
+          // interpolator. If it's using some other component, no sense in
+          // holding the volume data at this temporal id for this target.
+          // Second, if this isn't the temporal id we're interested in, then
+          // don't do anything.
```

Comment:
Make this clearer? (in particular "don't do anything" is not completely clear).  Maybe something like
```
Here we decide whether this interpolation target is "done" (i.e. it does not
need to interpolate) at this temporal_id. If it is "done", then we don't 
need to store the volume data. Usually "done" means that it has already
done its interpolation at this temporal_id. But note that if an interpolation
target is not using the Interpolator at all, it is considered "done", because
we don't need any volume data for it and should not store it.
Similarly, interpolation targets whose temporal_id has a different type
than TemporalId are considered "done" because they too do not use
the Interpolator and don't need volume data to be saved.
```